File: Sdk/Services/Dtos/ApiResult.cs
namespace Charlie.OpenIam.Sdk.Services.Dtos
{
    /// <summary>
    /// Api 调用结果
    /// </summary>
    public class ApiResult
    {
        /// <summary>
        /// 是否成功
        /// </summary>
        public bool IsSucceed { get; set; }

        /// <summary>
        /// Api 请求返回的状态码
        /// </summary>
        public int StatusCode { get; set; }

        /// <summary>
        /// 请求返回的消息，多用于当请求失败的信息
        /// </summary>
        public string Message { get; set; }
    }

    /// <summary>
    /// Api 调用结果
    /// </summary>
    /// <typeparam name="TData"></typeparam>
    public class ApiResult<TData>
    {
        /// <summary>
        /// 是否成功
        /// </summary>
        public bool IsSucceed { get; set; }

        /// <summary>
        /// Api 请求返回的状态码
        /// </summary>
        public int StatusCode { get; set; }

        /// <summary>
        /// Data 表示请求成功后的数据
        /// </summary>
        public TData Data { get; set; }

        /// <summary>
        /// 请求返回的消息，多用于当请求失败的信息
        /// </summary>
        public string Message { get; set; }
    }
}

File: Sdk/wwwroot/iam-client/src/ApiResult.ts
/**
 * Api 调用结果
 */
export default class ApiResult<TData = void> {

    /**
     * 是否成功
     */
    public IsSucceed: boolean;

    /**
     * Api 请求返回的状态码
     */
    public StatusCode?: number;

    /**
     * 请求返回的消息，多用于当请求失败的信息
     */
    public Message?: string;

    /**
     * 数据
     */
    public Data?: TData;
}

// export class ApiResultWithoutData {

//     /**
//      * 是否成功
//      */
//     public IsSucceed: boolean;

//     /**
//      * Api 请求返回的状态码
//      */
//     public StatusCode?: number;

//     /**
//      * 请求返回的消息，多用于当请求失败的信息
//      */
//     public Message?: string;
// }